lib/glific/templates.ex

```diff
@@ -171,8 +172,15 @@ defmodule Glific.Templates do
   @spec validate_button_template(map()) :: :ok | {:error, [String.t()]}
   defp validate_button_template(%{has_buttons: false} = _attrs), do: :ok
 
-  defp validate_button_template(%{has_buttons: true, button_type: _, buttons: _} = _attrs),
-    do: :ok
+  defp validate_button_template(%{has_buttons: true, button_type: _, buttons: buttons} = _attrs) do
+    invalid_texts = Enum.filter(buttons, fn %{"text" => text} ->
+      contains_invalid_chars?(text)
+    end)
+
+    if invalid_texts == [],
+      do: :ok,
+      else: {:error, ["Button Template", "Button texts contain invalid characters"]}
```

Comment:
Can we think of a more explicit error message that clearly states what is not allowed, such as emojis and other restrictions?